### PR TITLE
tests: enhance msg_avail

### PR DESCRIPTION
**Pull request by smlng**
- rename `thread_msg_avail` to `msg_avail`, does not require threads
    - minor code enhancements and corrections

### CODE REVIEW

File: tests/msg_avail/main.c [hunk 1]
```diff
@@ -34,28 +36,27 @@ int main(void)
 
     msg_init_queue(msg_queue, MSG_QUEUE_LENGTH);
 
+    puts("[START]");
```
Comment (kaspar030) [inline tests/msg_avail/main.c]: why puts here and log_info down there?

Comment (smlng) [inline tests/msg_avail/main.c]: I thought about making an issue (or mail on developer) regarding the general _structure_ of tests, but haven't done so for 😞 . My idea (RFC) would be to make tests more consistent in the overall workflow, that is:

- print/puts `[START]` when a tests begins and `[SUCCESS]`or `[FAILURE]` on completion. This should be always be output to allow for our test automation with `pexpect`, hence the `puts()`.
- any other output, which is only useful when running manually, should use `LOG_INFO` or `LOG_DEBUG`. So it can be omitted (or saved) especially for very constraint platforms, such as _arduino duemilanove` with only 2K RAM.
- every tests should also provide a (short) README on what it does, expected outcome. That way no test in-code description that blows up required RAM/flash sizes.
- and more ... 

I'll try to write this down soonish, what do you think?

**Comment by smlng** [comment]
ping @kaspar030

**Comment by smlng** [comment]
ping again

File: tests/msg_avail/tests/01-run.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import os
+import sys
+
+sys.path.append(os.path.join(os.environ['RIOTBASE'], 'dist/tools/testrunner'))
+import testrunner
+
+def testfunc(child):
+    child.expect_exact(u"[START]")
+    child.match
```
Comment (miri64) [inline tests/msg_avail/tests/01-run.py]: Why is this line here?

File: tests/msg_avail/main.c (hunk 1 above)
Comment (miri64) [inline tests/msg_avail/main.c]: > any other output, which is only useful when running manually, should use LOG_INFO or LOG_DEBUG. So it can be omitted (or saved) especially for very constraint platforms, such as _arduino duemilanove` with only 2K RAM.

This rule could make timing tests via pexpect impossible (which are due to missed UART timings hard to do already ;-)).

File: tests/msg_avail/tests/01-run.py (hunk 2 above)
Comment (smlng) [inline tests/msg_avail/tests/01-run.py]: öhm, don't remember exactly anymore - I think, I added it because without it the test ran through and was reported as successful even when `[FAILED]`was printed.

File: tests/msg_avail/main.c (hunk 1 above)
Comment (smlng) [inline tests/msg_avail/main.c]: mhm, yeah timing test might be a special thing to consider.  Still, currently many tests make excessive use of `puts` or `printf` which makes it impossible to fit those on constraint boards, i.e., like arduino uno or duemilanove with 2K RAM. And in most cases these `puts` could be replaced by a proper readme, or make output optional by using `LOG_<something>` instead.

Comment (smlng) [inline tests/msg_avail/main.c]: and more over, most output is only useful when running tests manually, but not if we (hopefully soon) integrate them into an automatic testing suite with the CI.

File: tests/msg_avail/tests/01-run.py (hunk 2 above)
Comment (miri64) [inline tests/msg_avail/tests/01-run.py]: But it does nothing... `child.match` is the Match object of the previous `expect` (not sure if `expect_exact` fills it, too), but you do nothing with it. Its like you would write

```py
None
```

in one line

Comment (smlng) [inline tests/msg_avail/tests/01-run.py]: I'll see into it, I simply want the test to fail if `[FAILED]` is printed xor the other way around to succeed if `[SUCCESS]` is printed.

Comment (smlng) [inline tests/msg_avail/tests/01-run.py]: so simply `child.expect_exact(u"[SUCCESS]")` should suffice, right?

Comment (miri64) [inline tests/msg_avail/tests/01-run.py]: Yes. If this does not appear in the output, child will throw a TIMEOUT exception.

Comment (smlng) [inline tests/msg_avail/tests/01-run.py]: well it works (see fixup) - I really cannot remember why I added those extra lines 😕

**Comment by miri64** [comment]
Please squash

**Comment by smlng** [comment]
squashed

**Comment by miri64** [comment]
I give @kaspar030 until next Friday to react. Otherwise I'm fine with the state of this PR.
